internal/cli/context_set.go

```diff
@@ -0,0 +1,99 @@
+package cli
+
+import (
+	"errors"
+
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+)
+
+type ContextSetCommand struct {
+	*baseCommand
+}
+
+func (c *ContextSetCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	flagSet := c.Flags()
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithNoConfig(),
+		WithClient(false),
+	); err != nil {
+		return 1
+	}
+
+	if c.flagWorkspace == "" {
+		// Require one argument
+		c.ui.Output(c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	contextName, err := c.contextStorage.Default()
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	if contextName == "" || contextName == "-" {
+		c.ui.Output(
+			clierrors.Humanize(errors.New("no default context exists")),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	// Load it and set it.
+	cfg, err := c.contextStorage.Load(contextName)
+	if err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// set new workspace
+	cfg.Workspace = c.flagWorkspace
+
+	// store updated context
+	if err := c.contextStorage.Set(contextName, cfg); err != nil {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	c.ui.Output("Context %q updated to use %s workspace.", contextName, cfg.Workspace, terminal.WithSuccessStyle())
```

Comment:
I think it's worth also including Server Address here, since most of the time context name will be pretty generic and hard to differentiate between servers (we do this on `waypoint status too).

Reply:
Good idea, I changed it in f0c2f42d to look like so:

```
$ waypoint context set -workspace dev

Context "install-1633109546" (localhost:9701) updated to use dev workspace.
```

Reply:
Looks good!